mllib/src/test/java/org/apache/spark/mllib/recommendation/JavaALSSuite.java

```diff
@@ -56,8 +56,7 @@ void validatePrediction(
       double matchThreshold,
       boolean implicitPrefs,
       DoubleMatrix truePrefs) {
-    List<Tuple2<Integer, Integer>> localUsersProducts =
-      Lists.newArrayListWithCapacity(users * products);
+    List<Tuple2<Integer, Integer>> localUsersProducts = new ArrayList();
```

Comment:
If you have to make any other changes on this PR, you could change one tiny thing here: preserve passing `users * products` as the initial capacity to this constructor.